plugins/net_plugin/include/eosio/net_plugin/net_plugin.hpp

```diff
@@ -20,10 +20,14 @@ namespace eosio {
       runtime_metric num_clients{metric_type::gauge, "num_clients", "num_clients", 0};
       runtime_metric dropped_trxs{metric_type::counter, "dropped_trxs", "dropped_trxs", 0};
 
-      net_plugin_metrics() {
-         metrics.emplace_back(std::ref(num_peers));
-         metrics.emplace_back(std::ref(num_clients));
-         metrics.emplace_back(std::ref(dropped_trxs));
+      virtual vector<runtime_metric> metrics() {
+         vector<runtime_metric> metrics;
+         metrics.reserve(3);
```

Comment:
Actually what is better so we don't have to maintain the reserve count is to do this:

```
         vector<runtime_metric> metrics{ 
            num_peers, 
            num_clients, 
            dropped_trxs 
         };
```